### PR TITLE
Feature/add reset password to login view

### PR DESCRIPTION
**Pull request by nunommts**
**What's this do?**
This PR adds a new link to handle on the authentication document regarding the user's password reset on some accounts. If that link is present on the library's authentication document, then a label is presented to the user so they can click on it and finish the process in a webview.

**Why are we doing this? (w/ JIRA link if applicable)**
There's [a feature request](https://www.notion.so/lyrasis/Android-Add-password-reset-link-for-virtual-library-cards-fd28cd2c5e0d480abf217a066ae94a09?p=d25f58e32a0444469382a700c0daf7fd&pm=c) to add this to the apps so they can support this feature.

**How should this be tested? / Do these changes have associated tests?**
Open the app
Select "LYRASIS Reads"
Go to settings and then to the account details screen
If you're logged in, logout
Confirm [the label is there](https://user-images.githubusercontent.com/79104027/181791136-408cc740-5d09-407d-9cd7-527bd7e856c8.png)
Click on it and verify [a webview's opened so you can reset your password](https://user-images.githubusercontent.com/79104027/181791226-4df817d8-dfe0-4204-be93-01e720580297.png)

**Dependencies for merging? Releasing to production?**
None

**Have you updated the changelog?**
Yes

**Has the application documentation been updated for these changes?**
No

**Did someone actually run this code to verify it works?**
Tested by @nunommts

### CODE REVIEW

File: simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountAuthenticationViewBindings.kt [hunk 1]
```diff
@@ -181,6 +190,19 @@ sealed class AccountAuthenticationViewBindings {
...
+    override fun setResetPasswordLabelStatus(
+      status: AccountLoginButtonStatus,
+      isVisible: Boolean,
+      onClick: () -> Unit
+    ) {
+      this.resetPasswordLabel.setOnClickListener {
+        onClick()
+      }
+      this.resetPasswordLabel.isVisible = isVisible && (
+        status is AsLoginButtonEnabled || status is AsLoginButtonDisabled
```
Comment (ray-lee) [inline simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountAuthenticationViewBindings.kt:202]: Why does the visibility of the password reset link depend on the status of the login button?

File: simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountAuthenticationViewBindings.kt [hunk 2]
```diff
@@ -330,6 +354,18 @@ sealed class AccountAuthenticationViewBindings {
...
+    override fun setResetPasswordLabelStatus(
+      status: AccountLoginButtonStatus,
+      isVisible: Boolean,
+      onClick: () -> Unit
+    ) {
+
+      this.resetPasswordLabel.setOnClickListener {
+        onClick()
+      }
+      this.resetPasswordLabel.isVisible = isVisible && status is AsLoginButtonEnabled
```
Comment (ray-lee) [inline simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountAuthenticationViewBindings.kt:366]: Same question.

**Review by ray-lee — COMMENTED** [review]
I just had a couple of questions below. Actually one question, two times.

File: simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountAuthenticationViewBindings.kt (hunk 1 above)
Comment (nunommts) [reply simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountAuthenticationViewBindings.kt:202]: Well, the feature says that the "forgot your password" label should appear when the user is not logged in, and those two status refer to that scenario. I assumed that as long as the login button is there, regardless of its state, the label should also be there (if there's an URI too). The remaining status refer to the "Cancel" and "Logout" options, so I thought the label shouldn't be visible in these cases

Comment (ray-lee) [reply simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountAuthenticationViewBindings.kt:202]: Thanks, that makes sense. Those states are confusing.

**Review by ray-lee — APPROVED** [review]

**Pull request merged**